### ISSUE TITLE
Empty folder remains on remote after deleting

### ISSUE THREAD

**Issue by mohanklein** [git-ftp/git-ftp#345]
When deleting a local folder (and commiting in local repo) and then pushing these changes to the ftp server the folder gets emptied on the server but remains there (empty) which causes the remote structure to get really messy tough it might not cause bad problems ...

**Comment by mkllnk** [comment]
Yes, that's a known issue. Git-ftp used to delete empty folders, but some FTP servers would delete folders even though they are not empty.

See also:

- https://github.com/git-ftp/git-ftp/issues/218
- https://github.com/git-ftp/git-ftp/issues/168

**Comment by mkllnk** [comment]
I think we could implement this. For all folders to be deleted, we need to check if the folder is empty and only send the delete command if it's empty. Any volunteers?

**Comment by LukasFritzeDev** [comment]
Sorry. I thought git only tracks files, not folders. So does git-ftp really calls a delete command for folders?

**Comment by mkllnk** [comment]
Git tracks only files, that's correct. But if you check out a branch where all files from a folder have been deleted then Git deletes that folder as well. It does that only if the folder is empty. Of course it keeps folders that have other untracked files in there.

Git-ftp did the same once. But since the FTP rmdir command is supposed to delete only empty folders, Git-ftp didn't bother checking for emptiness. It would just send the delete command and only empty folders got deleted. Well, until someone reported that some servers delete the folder anyway, with all files in there. So we switched that feature off.

This issue is now about re-introducing the feature to delete empty directories if they became empty due to the push. It requires checking those directories for emptiness first.

**Comment by rizkysyazuli** [comment]
Yes, please implement this feature. This has been causing problems actually for file-based CMS. So it's not just a matter of tidyness.

Right now, I'm using Statamic. It creates a content structure like this.

```
- 1. home
  -- index.md
- 2. about
  -- index.md
- 3. contact
  -- index.md
```

FYI: The folder names represent the page name, the numbers reflects the page ordering on the front end, and the content itself is a markdown file inside each folders.

But let say one day i decided to remove the "about" page. Locally, it will look like this:

```
- 1. home
  -- index.md
- 2. contact
  -- index.md
```

But since Git does not remove the folder on the remote server.. It ends up like this:

```
- 1. home
  -- index.md
- 2. about
  -- (empty)
- 2. contact
  -- index.md
- 3. contact
  -- (empty)
```

Now i have two "contact" folders. I think this confuses Statamic. Because when i edit the "contact" page, the changes sometimes is saved to the wrong folder. And the front end is not updated.

I'm currently considering either: a) switch to rsync b) make my remote into a git repo as well

**Comment by ecxod** [comment]
please do not close this.
> But if you check out a branch where all files from a folder have been deleted then Git deletes that folder as well.

This is exactly what is expected and does not happen.

**Comment by jonathan-reisdorf** [comment]
We're running into the same issue when using git-ftp to sync content files for Kirby CMS (a file-based CMS). Empty folders in this case confuse the CMS because essentially they are empty pages. Is there an option to enforce deleting empty folders or a fork that accomplishes this?
